vhdbuilder/packer/cleanup.sh

```diff
@@ -181,7 +181,58 @@ if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" && "${DRY_RUN,,}" == "false" ]]; then
   set -x
 fi
 
-#clean up storage account created over a week ago
+# attempt to clean up Windows managed images and SIG image versions created over a week ago in SIG_GALLERY_NAME (except the production gallery)
+if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" && "${DRY_RUN}" == "False" && "${SIG_GALLERY_NAME}" != "AKSWindows" ]]; then
+  echo "Looking for Windows managed images in ${AZURE_RESOURCE_GROUP_NAME} created over ${EXPIRATION_IN_HOURS} hours ago..."
+
+  managed_image_ids=""
+  sig_version_ids=""
+
+  # delete outdated Windows managed images and associated SIG versions (.tags.os must be "Windows")
+  images=$(az image list -g ${AZURE_RESOURCE_GROUP_NAME} | jq --arg dl $deadline '.[] | select(.tags.os == "Windows") | select(.tags.now < $dl).name' | tr -d '\"' || "")
+  for image in $images; do
+    managed_image_ids="${managed_image_ids} /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP_NAME}/providers/Microsoft.Compute/images/${image}"
+    sig_version_ids="${sig_version_ids} /subscriptions/${SUBSCRIPTION_ID}/resourceGroups/${AZURE_RESOURCE_GROUP_NAME}/providers/Microsoft.Compute/galleries/${SIG_GALLERY_NAME}/images/${image%-*}/versions/${image#*-}"
+    echo "Will delete Windows managed image ${image} and associated SIG version from resource group ${AZURE_RESOURCE_GROUP_NAME}"
+  done
+
+  if [[ -n "${managed_image_ids}" ]]; then
+    echo "Attempting to delete $(echo ${managed_image_ids} | wc -w) Windows managed images..."
+    az resource delete --ids ${managed_image_ids} > /dev/null || echo "Windows managed image deletion was not successful, continuing..."
+  else
+    echo "Did not find any Windows managed images eligible for deletion"
+  fi
+
+  if [[ -n "${sig_version_ids}" ]]; then
+    echo "Attempting to delete $(echo ${sig_version_ids} | wc -w) Windows SIG image versions associated with old managed images..."
+    az resource delete --ids ${sig_version_ids} > /dev/null || echo "Windows SIG image version deletion was not successful, continuing..."
+  else
+    echo "Did not find any SIG versions associated with old Windows managed images eligible for deletion"
+  fi
+fi
+
+# attempt to clean up Windows SIG image versions in all galleries except "AKSWindows" created over a week ago
+if [[ -n "${AZURE_RESOURCE_GROUP_NAME}" && "${DRY_RUN}" == "False" ]]; then
+  gallery_lists=$(az sig list -g ${AZURE_RESOURCE_GROUP_NAME} | jq '.[].name' | tr -d '\"' || "")
+  for gallery in $gallery_lists; do
+    if [[ $gallery == "AKSWindows" ]]; then
+        continue
+    fi
+
+    # delete old Windows SIG image versions in gallery (image definitions must have .osType == "Windows")
+    image_defs=$(az sig image-definition list -g ${AZURE_RESOURCE_GROUP_NAME} -r ${gallery} | jq '.[] .name' | tr -d '\"' || "")
+    for image_definition in $(az sig image-definition list -g ${AZURE_RESOURCE_GROUP_NAME} -r ${gallery} | jq '.[] | select(.osType == "Windows").name' | tr -d '\"' || ""); do
```

Comment:
I think that we should avoid list all image-definition in every for loop.
1. It will increase the finish time
2. it may not exit if some PRs are being submitted to create new image definitions continously.